src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs

```diff
@@ -81,6 +81,13 @@ public static IResourceBuilder<ContainerResource> AddContainer(this IDistributed
         return builder.WithAnnotation(annotation);
     }
 
+    /// <summary>
+    /// Adds the arguments to be passed to a container resource when the container is started.
```

Comment:
This extension method isn't currently constrained to `ContainerResource` (see line 91 below) so I'm not sure what it does when used with other resource types. We'll likely need to check that and update the doc comments appropriately.
@eerhardt @davidfowl

Reply:
I had a look through the implementation. It is definitely container specific.

Reply:
It's not specific right now. It also works for executables.

Reply:
@mitchdenny @davidfowl so, which is it? Constraining this method to `where T : ContainerResource` seems wrong if this indeed works for executables too.

Reply:
It's both executables and containers.

Reply:
David is correct. I must have been hallucinating.

Reply:
Actually I don't think that `WithArgs` is fully baked. I am wondering if it was actually intended to be private. @BrennanConroy

Reply:
https://github.com/dotnet/aspire/blob/91034428ea2f6058ed572920059b680e7d7cdcc0/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs#L572-L579

Reply:
OK find references in VS might be having an issue ;)